crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net_task.rs

```diff
@@ -1880,6 +2343,42 @@ mod tests {
             .unwrap()
     }
 
+    async fn verify_task_live(
+        ep_evt_send: Sender<MetaNetEvt>,
+        meta_net_task_finished: Arc<AtomicBool>,
+    ) {
+        assert!(!meta_net_task_finished.load(Ordering::Acquire));
+
+        let response = do_request(
+            ep_evt_send,
+            wire::Wire::PeerQuery(wire::PeerQuery {
+                space: test_space(1),
+                basis_loc: DhtLocation::new(1),
+            }),
+        )
+        .await;
+
+        let peer_list = match response {
+            wire::Wire::PeerQueryResp(r) => r.peer_list,
+            r => panic!("Unexpected response - {:?}", r),
+        };
+
+        assert_eq!(8, peer_list.len());
+        assert!(!meta_net_task_finished.load(Ordering::Acquire));
+    }
+
+    async fn wait_and_assert_shutdown(meta_net_task_finished: Arc<AtomicBool>) {
+        tokio::time::timeout(Duration::from_millis(1000), async {
+            while !meta_net_task_finished.load(Ordering::Acquire) {
+                tokio::time::sleep(Duration::from_millis(1)).await;
+            }
+        })
+        .await
+        .expect("Timed out waiting for shutdown");
+
+        assert!(meta_net_task_finished.load(Ordering::Acquire));
+    }
```

Comment:
If this took a closure, you could clean up the other instances using this same pattern

```suggestion
    async fn wait_and_assert_shutdown(finished: impl Fn() -> bool) {
        tokio::time::timeout(Duration::from_millis(1000), async {
            while !finished() {
                tokio::time::sleep(Duration::from_millis(1)).await;
            }
        })
        .await
        .expect("Timed out waiting for shutdown");

        assert!(finished());
    }
```